### PR TITLE
fix: [Violations] Distance - Incorrect error message when distance amount is changed to smaller amount.

### PR DESCRIPTION
**Pull request by Krishna2323**
### Fixed Issues
$ https://github.com/Expensify/App/issues/41401
PROPOSAL: https://github.com/Expensify/App/issues/41401#issuecomment-2091932028

### Tests
1. Go to Submit expense > Scan > Add valid receipt > Select workspace > Submit
2. Wait for scanning to complete, once completed edit the amount to a higher value
3. Verify `Amount greater than scanned receipt` error appears below amount field 
4. Go to Submit expense > Distance > Select 2 waypoints > Select workspace > Submit
5. Open transaction page > Wait for amount to be calculated > Edit the amount to higher value 
6. Verify `Amount differs from calculated distance` error appears below amount field 

- [x] Verify that no errors appear in the JS console

### Offline tests
1. Go to Submit expense > Scan > Add valid receipt > Select workspace > Submit
2. Wait for scanning to complete, once completed edit the amount to a higher value
3. Verify `Amount greater than scanned receipt` error appears below amount field 
4. Go to Submit expense > Distance > Select 2 waypoints > Select workspace > Submit
5. Open transaction page > Wait for amount to be calculated > Edit the amount to higher value 
6. Verify `Amount differs from calculated distance` error appears below amount field 

### QA Steps
1. Go to Submit expense > Scan > Add valid receipt > Select workspace > Submit
2. Wait for scanning to complete, once completed edit the amount to a higher value
3. Verify `Amount greater than scanned receipt` error appears below amount field 
4. Go to Submit expense > Distance > Select 2 waypoints > Select workspace > Submit
5. Open transaction page > Wait for amount to be calculated > Edit the amount to higher value 
6. Verify `Amount differs from calculated distance` error appears below amount field 

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

distance_issue
scan_issue

</details>

<details>
<summary>Android: mWeb Chrome</summary>

distance_issue
scan_issue

</details>

<details>
<summary>iOS: Native</summary>

distance_issue
scan_issue

</details>

<details>
<summary>iOS: mWeb Safari</summary>

distance_issue
scan_issue

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

distance_amount_violation
receipt_scan_issue

</details>

<details>
<summary>MacOS: Desktop</summary>

distance_issue
scan_issue

</details>

### CODE REVIEW

**Review by sobitneupane — COMMENTED** [review]
### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>


Screenshot 2024-05-07 at 14 39 15
Screenshot 2024-05-07 at 14 40 08
Screenshot 2024-05-07 at 14 41 08


</details>

<details>
<summary>iOS: mWeb Safari</summary>


Screenshot 2024-05-07 at 14 34 06


</details>

<details>
<summary>MacOS: Chrome / Safari</summary>


https://github.com/Expensify/App/assets/25876548/3bc60772-783e-4bb9-b0b4-925958146ed7

Screenshot 2024-05-07 at 14 16 50
Screenshot 2024-05-07 at 14 16 59
Screenshot 2024-05-07 at 14 17 16
Screenshot 2024-05-07 at 14 20 10

</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Comment by sobitneupane** [comment]
@Krishna2323 Have we considered different types as mentioned [here](https://github.com/Expensify/App/issues/41401#issuecomment-2091898713)? Can you please complete the PR review checklist. Please include the test cases for different types.

**Comment by JmillsExpensify** [comment]
Whoops, sorry missed this. 

> One more question on the above comment: do we also need to remove the taxAmountChanged notice, along with taxRateChanged, from NewDot?

Yes, that's correct. We aren't going to show these on NewDot, as both will generate system messages when changed. We also show what tax rate is "default" in a parallel initiative that's currently in implementation for track tracking in New Expensify.

**Comment by JmillsExpensify** [comment]
Just confirming that the above response makes sense?

**Comment by sobitneupane** [comment]
> However, I'm not sure what to add in the test cases. Could you provide some guidance? Several types of amount notices are there.

@Krishna2323 You can include the test cases based on [this expected output](https://github.com/Expensify/App/issues/41401#issuecomment-2091898713).

Verify that the notice violations related to Amount is shown below the amount field.
> Let's show all these cases under the amount field, since a) we think it's more impactful/intuitive to highlight that the amount is greater than the receipt, than the other way around; and b) the field you need to take action on in order to fix the violation is the amount field, not the receipt field

Verify notice violations based on their type
> {data: {type: 'distance'}}
{data: {type: 'card'}}
{data: {type: 'smartscan'}}
{data: {type: 'smartscan', displayPercentVariance: $displayPercentVariance}}

You can also include Tests for the bug you discussed [here](https://github.com/Expensify/App/pull/41649#issuecomment-2094504645).

Basically, the tests should cover the changes made in the PR. You can add multiple set of test cases.

**Comment by sobitneupane** [comment]
In distance request, I can still see the wrong error message.

Screenshot 2024-05-20 at 12 58 48

**Comment by JmillsExpensify** [comment]
I think you mean @cead22

**Comment by cead22** [comment]
Yes, I'll get the PR for that submitted today/tomorrow

**Comment by cead22** [comment]
I submitted the PR for review

**Comment by cead22** [comment]
Yeah it's live!

**Comment by sobitneupane** [comment]
@Krishna2323 Did you test the PR in your end? Please make sure to test the PR in your end first.

**Comment by sobitneupane** [comment]
> In the screenshot below, we show receipt verified when the amount field has an issue. Should we be showing issue found instead?

Yes, I agree. We should be showing 'issue found' instead.

**Review by sobitneupane — COMMENTED** [review]
### Screenshots/Videos
<details>
<summary>MacOS: Chrome / Safari</summary>
Screenshot 2024-06-14 at 13 27 11
Screenshot 2024-06-14 at 13 30 05
Screenshot 2024-06-14 at 13 31 51
Screenshot 2024-06-14 at 13 32 52
Screenshot 2024-06-14 at 13 33 13
Screenshot 2024-06-14 at 13 38 08
Screenshot 2024-06-14 at 13 57 29



</details>
<details>
<summary>Android: Native</summary>
Screenshot 2024-06-14 at 14 32 11
Screenshot 2024-06-14 at 14 32 20
Screenshot 2024-06-14 at 14 32 31
Screenshot 2024-06-14 at 14 32 54
Screenshot 2024-06-14 at 14 37 07



</details>

<details>
<summary>iOS: Native</summary>
Screenshot 2024-06-14 at 14 29 35
Screenshot 2024-06-14 at 14 30 09
Screenshot 2024-06-14 at 14 30 25
Screenshot 2024-06-14 at 14 30 35



</details>

<details>
<summary>iOS: mWeb Safari</summary>
Screenshot 2024-06-14 at 14 19 28
Screenshot 2024-06-14 at 14 19 40
Screenshot 2024-06-14 at 14 19 49
Screenshot 2024-06-14 at 14 19 59



</details>

File: src/languages/en.ts [hunk 1]
```diff
@@ -3132,7 +3133,19 @@ export default {
...
+                    return 'Amount differs from calculated distance';
+                case 'card':
+                    if (displayPercentVariance) {
+                        return `Amount ${displayPercentVariance}% greater than scanned receipt`;
+                    }
+                    return 'Amount greater than card transaction';
+                default:
+                    return 'Amount greater than scanned receipt';
+            }
+        },
```
Comment (sobitneupane) [inline src/languages/en.ts:3148]: https://github.com/Expensify/App/issues/41401#issuecomment-2091898713

>   * `{data: {type: 'distance'}}`
>   * `{data: {type: 'card'}}`
>   * `{data: {type: 'smartscan'}}`
>   * `{data: {type: 'smartscan', displayPercentVariance: $displayPercentVariance}}`

`displayPercentVariance` is present in type 'smartscan' not 'card'

**Comment by sobitneupane** [comment]
I could only verify the first and the third cases.

>{data: {type: 'distance'}}
{data: {type: 'card'}}
{data: {type: 'smartscan'}}
{data: {type: 'smartscan', displayPercentVariance: $displayPercentVariance}}

@cead22 @JmillsExpensify  How can I test the 2nd and the 4th cases?

**Comment by cead22** [comment]
To get the violation with `{data: {type: 'card'}}`
- Import a transaction from a personal credit card
- Manually update the amount to be more than 5% higher than the original amount

To get the violation with `{data: {type: 'smartscan', displayPercentVariance: $displayPercentVariance}}`
- Create a transaction by scanning a receipt
- Once it finished scanning, change the currency to one with similar value (eg, EUR <> USD <> GPB, and not USD => IDR), and change amount to be twice as much as the original

Let me know if that doesn't work, and maybe we can test this when we deploy the app with some staging/production data

**Comment by sobitneupane** [comment]
Thank you, @cead22. I am not sure if I will be able to test the first case, but I will test the second case. In the meantime, @Krishna2323, can you please test it on your end as well?

**Comment by sobitneupane** [comment]
I am also seeing some case where we show "Duplicate" below the "Merchant" field as an issue. I am not sure it is expected as there is not much information about the issue, it is just "Duplicate". 

cc: @JmillsExpensify @cead22

**Comment by cead22** [comment]
@sobitneupane can you share a screenshot and reproduction steps for that please?

**Comment by sobitneupane** [comment]
@cead22 I am not very sure how to reproduce. But I am pretty sure it was introduced by this [commit](https://github.com/Expensify/App/pull/41649/commits/407e5522bc97ca07a690b8c3f6cef3b4e10cebdf).  

I believe it is returned as warning type violations as mentioned by @Krishna2323 [here](https://github.com/Expensify/App/pull/41649#issuecomment-2154162701).


> I see that we now also get warning type violations. I think these should be included in the paid policy groups as well, correct? (I've already updated the code for this.)

Screenshot 2024-06-18 at 13 18 18

**Comment by cead22** [comment]
> I don't know how to add a personal credit card. Do I need to add a real credit card?

I think you can skip this one and we can test it when it's on staging. And you can ignore the duplicate one for now

**Comment by sobitneupane** [comment]
@Krishna2323 You can select US location from developers option > Sensor > Location in the browser.

**Comment by sobitneupane** [comment]
>To get the violation with {data: {type: 'smartscan', displayPercentVariance: $displayPercentVariance}}
>- Create a transaction by scanning a receipt
>- Once it finished scanning, change the currency to one with similar value (eg, EUR <> USD <> GPB, and not USD => IDR), and change amount to be twice as much as the original

It seems we should be using `violation.data.displayPercentVariance` for `displayPercentVariance`. However, the code is currently using `violation.displayPercentVariance`. 

Despite following the steps, I couldn't retrieve the data in the format {type: 'smartscan', displayPercentVariance: $displayPercentVariance}.

Screenshot 2024-06-20 at 13 42 22

https://github.com/Expensify/App/assets/25876548/b1c8fcca-de7e-47cc-bff6-fa6d96e5a4a1

**Comment by sobitneupane** [comment]
> I tried after changing the location from Sensor but the currency is still INR.

Can you please try it again after refreshing the page. It should work. It is working for me.

**Comment by sobitneupane** [comment]
@Krishna2323 It is working well for me. I am not sure why it is not working in your end. I am using configurations as shown in [screen recording here.](https://github.com/Expensify/App/pull/41649#issuecomment-2180063894) You might have to use VPN if it's not working for you.

**Comment by cead22** [comment]
> It seems we should be using `violation.data.displayPercentVariance`

This is correct, at least in the sense that `displayPercentVariance` should be in the violation `data`

**Comment by cead22** [comment]
Do you have an ETA for when this will be ready?

**Comment by cead22** [comment]
Thanks

> This is the last test we need to pass and it seems like the backend isn't providing `displayPercentVariance`.

If you confirm this is the case, then we can handle it in a follow up PR

**Comment by cead22** [comment]
I've been trying to reproduce this myself and get a violation with `displayPercentVariance` and haven't been able to, so maybe this  can't be reproduced in new dot, so let's not block on it

**Comment by cead22** [comment]
> @cead22, ok I will try once more, don't know why the currency automatically changes to INR in my case.

I'm pretty sure we set the currency based on IP address!

**Comment by cead22** [comment]
- @Krishna2323 how did your review go? 
- @sobitneupane can you do another review today?

**Comment by cead22** [comment]
@sobitneupane bump please

**Review by sobitneupane — COMMENTED** [review]
### Screenshots/Videos
<details>
<summary>Android: Native</summary>
Screenshot 2024-07-03 at 12 57 51
Screenshot 2024-07-03 at 12 58 59
Screenshot 2024-07-03 at 13 02 22
Screenshot 2024-07-03 at 13 03 35
Screenshot 2024-07-03 at 13 06 27

https://github.com/Expensify/App/assets/25876548/31f17c51-a33c-4d6c-8c19-ec5decc5e393

</details>

<details>
<summary>Android: mWeb Chrome</summary>

Screenshot_20240704-182730_Chrome

Screenshot_20240704-190444_Chrome
Screenshot_20240704-190519_Chrome
Screenshot_20240704-191411_Chrome
Screenshot_20240704-191729_Chrome


</details>
<details>
<summary>iOS: Native</summary>

Screenshot 2024-07-03 at 12 43 36
Screenshot 2024-07-03 at 12 44 29
Screenshot 2024-07-03 at 12 45 44
Screenshot 2024-07-03 at 12 46 40
Screenshot 2024-07-03 at 12 47 39
Screenshot 2024-07-03 at 12 54 58

</details>

<details>
<summary>iOS: mWeb Safari</summary>


https://github.com/Expensify/App/assets/25876548/5ebb4fab-6a23-4789-9412-2fb9010df4c4


https://github.com/Expensify/App/assets/25876548/a81b814e-3819-4220-b695-e1ca0e2978c0


</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/25876548/78028b6e-10d7-4c03-9256-c54c5b302b0a


https://github.com/Expensify/App/assets/25876548/eeadab60-bd77-4325-9667-6be0741305cb

Screenshot 2024-07-03 at 12 19 04
Screenshot 2024-07-03 at 12 20 22
Screenshot 2024-07-03 at 12 21 38
Screenshot 2024-07-03 at 12 23 29
Screenshot 2024-07-03 at 12 23 47
Screenshot 2024-07-03 at 12 26 07
Screenshot 2024-07-03 at 12 29 37



</details>
<details>
<summary>MacOS: Desktop</summary>

Screenshot 2024-07-04 at 17 47 00
Screenshot 2024-07-04 at 17 47 28
Screenshot 2024-07-04 at 17 47 34
Screenshot 2024-07-04 at 17 47 41
Screenshot 2024-07-04 at 17 47 59
Screenshot 2024-07-04 at 17 48 57
Screenshot 2024-07-04 at 17 49 03

</details>

**Review by sobitneupane — APPROVED** [review]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

File: src/components/ReportActionItem/MoneyRequestView.tsx [hunk 2]
```diff
@@ -390,9 +391,10 @@ function MoneyRequestView({
             {shouldShowAnimatedBackground && <AnimatedEmptyStateBackground />}
             <>
                 {shouldShowReceiptHeader && (
-                    <ReceiptAuditHeader
+                    <ReceiptAudit
                         notes={receiptViolations}
-                        shouldShowAuditMessage={!!(shouldShowNotesViolations && didRceiptScanSucceed)}
+                        shouldShowAuditSuccess={shouldShowAuditMessage && didRceiptScanSucceed}
+                        shouldShowAuditFailure={shouldShowAuditMessage && hasReceipt}
```
Comment (cead22) [inline src/components/ReportActionItem/MoneyRequestView.tsx:397]: I've been looking at this for a minute, and it's not obvious to me why we now need shouldShowAuditSuccess and shouldShowAuditFailure, instead of using notes.length to determine what to show.

What cases would we miss if we went back to relying only on notes.length? If you can update the docs for these params to make it clear, please do

**Review by cead22 — CHANGES_REQUESTED** [review]

**Comment by sobitneupane** [comment]
Bump @Krishna2323. Can you please prioritize this PR.

**Comment by sobitneupane** [comment]
@Krishna2323 The violation message are being shown below the Receipt instead of respective field.
Screenshot 2024-07-08 at 14 52 09

File: src/components/ReportActionItem/MoneyRequestView.tsx (hunk 2 above)
Comment (cead22) [reply src/components/ReportActionItem/MoneyRequestView.tsx]: Can you confirm, and if you do, add a manual tests that would reproduce the issue, and confirm your change solves it. If you can't confirm, let's revert to what we had previously

File: src/components/ReportActionItem/MoneyRequestView.tsx [hunk 3]
```diff
@@ -448,7 +450,7 @@ function MoneyRequestView({
                     />
                 )}
                 {!shouldShowReceiptEmptyState && !hasReceipt && <View style={{marginVertical: 6}} />}
-                {shouldShowNotesViolations && <ReceiptAuditMessages notes={receiptViolations} />}
+                {canUseViolations && <ViolationMessages violations={getViolationsForField('receipt')} />}
```
[the remark this answers is not in the document]
Comment (cead22) [reply src/components/ReportActionItem/MoneyRequestView.tsx]: That change must have broken something that isn't being tested in the manual tests of this PR (otherwise we would've noticed it broke something). Given that, consider adding a manual test and confirming your changes/code solve for the case in question

**Comment by cead22** [comment]
> @Krishna2323 The violation message are being shown below the Receipt instead of respective field.

That means we might not be using `violationFields` or the flow that uses `violationFields` to get the mapping of _violation -> field to show violation_

File: src/components/ReceiptAudit.tsx [hunk 4]
```diff
@@ -7,17 +7,34 @@ import Icon from './Icon';
...
+    notes: string[];
+
+    /** Weather to show success message (e.g. When receipt is scanning, we don't want to show `Verified` message) */
+    shouldShowAuditSuccess: boolean;
+
+    /** Weather to show failure message (e.g. When receipt isn't required, we don't want to show `Issues Found` message) */
+    shouldShowAuditFailure: boolean;
+};
+
+function ReceiptAudit({notes, shouldShowAuditSuccess, shouldShowAuditFailure}: ReceiptAuditPropsType) {
```
Comment (cead22) [inline src/components/ReceiptAudit.tsx:21]:
```suggestion
type ReceiptAuditProps = {
    /** List of audit notes */
    notes: string[];

    /** Wether to show success message (e.g. When receipt is scanning, we don't want to show `Verified` message) */
    shouldShowAuditSuccess: boolean;

    /** Wether to show failure message (e.g. When receipt isn't required, we don't want to show the `Issues Found` message) */

function ReceiptAudit({notes, shouldShowAuditSuccess, shouldShowAuditFailure}: ReceiptAuditProps) {
```

File: src/components/ReportActionItem/MoneyRequestView.tsx [hunk 5]
```diff
@@ -389,10 +390,11 @@ function MoneyRequestView({
         <View style={styles.pRelative}>
             {shouldShowAnimatedBackground && <AnimatedEmptyStateBackground />}
             <>
-                {shouldShowReceiptHeader && (
-                    <ReceiptAuditHeader
+                {shouldShowReceiptAudit && (
+                    <ReceiptAudit
                         notes={receiptViolations}
-                        shouldShowAuditMessage={!!(shouldShowNotesViolations && didRceiptScanSucceed)}
+                        shouldShowAuditSuccess={!!(shouldShowAuditMessage && didRceiptScanSucceed)}
```
Comment (cead22) [inline src/components/ReportActionItem/MoneyRequestView.tsx:396]: Can we make these variables Boolean so we don't need the casting outside?

File: src/hooks/useViolations.ts [hunk 6]
```diff
@@ -40,26 +40,42 @@ const violationFields: Record<ViolationName, ViolationField> = {
...
 type ViolationsMap = Map<ViolationField, TransactionViolation[]>;
 
-function useViolations(violations: TransactionViolation[]) {
+// We don't want to show these violations on NewDot
+const excludedViolationsName = ['taxAmountChanged', 'taxRateChanged'];
+
+/**
+ * @param violations – List of transaction violation
+ * @param shouldShowOnlyViolations – Whether we should only show violations of type 'violation
+ */
```
Comment (cead22) [inline src/hooks/useViolations.ts:58]:
```suggestion
 * @param violations – List of transaction violations
 * @param shouldShowOnlyViolations – Whether we should only show violations of type 'violation'
 */
```

**Review by cead22 — CHANGES_REQUESTED** [review]

**Comment by cead22** [comment]
@Krishna2323 please put this at the top of your priority list so we can merge this this week (ideally today/tomorrow)

**Comment by cead22** [comment]
@Krishna2323 what about https://github.com/Expensify/App/pull/41649#discussion_r1668942122 ?

**Comment by sobitneupane** [comment]
@Krishna2323 Please make sure to test the PR in your end after each change.

File: src/components/ReportActionItem/MoneyRequestView.tsx (hunk 2 above)
Comment (cead22) [reply src/components/ReportActionItem/MoneyRequestView.tsx:395]: @Krishna2323 sorry if this is obvious, but can you explain why we can't use a single prop? Both success and failure need to be shown only if there's a receipt, and the issue you linked shows an error when there's no receipt. What am I missing? Can you share test steps to reproduce the issue when we use only 1 prop, that doesn't happen with this solution?

Comment (cead22) [reply src/components/ReportActionItem/MoneyRequestView.tsx:395]: I think I get it now after another look at the code, sorry about that. The thing is we want to show
1. Nothing if there's no receipt and one can't be added
2. "Receipt" if there is no receipt, but one can be added (receipt empty state would show below this)
3. "Receipt * Issue found" if there's a receipt, and there are notes
4. "Receipt * Verified" if there's a receipt, and no notes

So what about doing something like below, and we get rid of `shouldShowAuditSuccess` & `shouldShowAuditFailure`?

```
const shouldShowReceiptAudit = isReceiptAllowed && (shouldShowReceiptEmptyState || hasReceipt);

// New
// Set receiptViolations to empty so no audit messages are shown
if (!shouldShowAuditMessage || !didRceiptScanSucceed) {
    receiptViolations = [];
}
```

Then for the cases above
1. We'd show nothing if there's no receipt and one can't be added based on `{shouldShowReceiptAudit &&<ReceiptAudit ...` and the fact that `shouldShowReceiptAudit` would be false
2. "Receipt" since `shouldShowReceiptAudit` would be true and the ReceiptAudit component always renders "Receipt", and we'd set `receiptViolations` (`notes` inside the component) to empty array because there's no receipt and `didRceiptScanSucceed` would be false
3. "Receipt * Issue found" since `shouldShowReceiptAudit` would be true, and  both `shouldShowAuditMessage` and `didRceiptScanSucceed` would be true, so we won't set `receiptViolations` to empty array
4. "Receipt * Verified" since `shouldShowReceiptAudit` would be true, and `receiptViolations`/`notes` will already be empty

Comment (cead22) [reply src/components/ReportActionItem/MoneyRequestView.tsx]: Love it, sorry for the late reply

Comment (cead22) [reply src/components/ReportActionItem/MoneyRequestView.tsx]: @Krishna2323 will you be able to update this today and get the PR ready for review?

**Review by cead22 — APPROVED** [review]
